analysis/runtime/src/events.rs

```diff
@@ -30,10 +30,8 @@ pub enum EventKind {
 
     CopyRef,
 
-    /// Field projection. Used for operations like `_2 = &(*_1).0`. Nested field
-    /// accesses like `_4 = &(*_1).x.y.z` are broken into multiple `Node`s, each
-    /// covering one level.
-    Field(Pointer, u32),
+    /// Projection. Used for operations like `_2 = &(*_1).0`.
+    Project(Pointer, Pointer),
```

Comment:
Can you update the comment to say more about the meaning of the two `Pointer` fields? I guess the first one is the base pointer and the second is the result of the projection?

Also, since the text "Nested field accesses are broken into multiple `Node`s" was removed, does that mean a single `Project` might project through multiple levels of structs?  If so, is there any ambiguity in cases like this:

```Rust
struct A {
  x: i32,
  y: i32,
}
struct B {
  z: i32,
  a: A,
}
let p: *const B = ...;
// Suppose p == 0x1000
let q1: *const A = &(*p).a;
// q1 == 0x1004
let q2: *const i32 = &(*p).a.x;
// q2 == 0x1004
```
Note that this distinction matters to the borrow checker in some cases:
```Rust
let mut b: B = ...;
let r1 = &mut b.a.y;
let r2 = &mut b.a.x; // Legal
let r3 = &mut b.a; // Same address as r2, but not legal
*r1 = 1;
```

Reply:
Right, the current encoding does not distinguish sub-fields at offset 0 from their parents. I couldn't think of a case where it mattered, thanks for the example!

We can't break down nested field accesses into multiple events (the old implementation was broken and required some hackiness in the PDG), so I had an idea for an alternative: give each combination of nested subfields its own unique ID, and add that to the event. In your example above, `b.a.y` would get ID 0, `b.a.x` would be 1, and `b.a` would be 2. Then `b.a.x` and `b.a` could be differentiated by that number.

Reply:
> `b.a.y` would get ID 0, `b.a.x` would be 1, and `b.a` would be 2. Then `b.a.x` and `b.a` could be differentiated by that number.

As long as there's a lookup table somewhere that can map those IDs back to paths (e.g. 0 -> `.a.y`), that sounds fine - it would provide the same information as the nested version.

Reply:
I added that, and moved it into #1107.